VA-Explorer/va_explorer
# Issue132 - Improve UI/UX of editting a verbal autopsy record

closes #132 

### Motivation: 
The UI/UX of editing a record as it stands has room for improvement. For starters, every single field has a text field as input. This is a hinderance as many of the questions are not open ended and require a specific answer choice from a select set of choices.

### How I found the which model questions needed a radio field:
I did not go in and read every field. I went into an example VA found in our resources and that for each question, there was a specified "type" of answer. This includes integer, text, and the one I was looking for **"select_one."** Many of these fields had `select_one YES_NO_DK_REF` or something similar, so I used this as a guide of which fields needed a radio field.

To get every model field variable name, I did not copy and paste every single variable name. Because there were so many, I wrote a short python script where I inputted all the lines of code where we created fields for our VerbalAutopsy model. What I expected as an output, was a clean list of all our of field variable names. 
For here, I did start working by hand, where I would look at the excel sheet and find which fields had `select_one [x]`, and I would append it to a dictionary in `models.py` called RADIO_CHOICES where I map it to the choice set it needs.
 
### Room for improvement after this PR:

- There are still sets of choices besides "YES_NO_DK_REF" that are mostly used only once or a few times. I only included one, `MALE_FEMALE` because it was a small choice set. The problem with other sets is they can be large and I wasn't sure what possible choices are in the set, which is why I left them out. For example, for the question "Where did the deceased die?", the VA type expects `select_one select_18`, which is presumably a choice set of 18 types of locations.
- Some model questions require integers as input, so some of these fields should use `IntegerField` instead of `CharField`
- The UI is clean, but can probably be made to look nicer. This task would take a lot of time of designing and implementing to get to a good point
- Field value placeholders were done on there own somehow (probably some Django field default behavior). And because of [this PR code change](https://github.com/VA-Explorer/va_explorer/pull/184), sometimes the field shown on the view template ( (`show/#`), does not match the text field placeholder in the edit page (`edit/#`). This is because the implementation of view page applies a filter to some text results, while the edit page seems to automatically create text field placeholders by pulling the actual value from the DB.
An example of this is that `nan` will appear on some field placeholders, even though we got rid of instances of `nan` in the VA show page because of its unclearness.


### Testing instructions:

1.) Load up data_management tab with mock data installed.
2.) Click "View" for any row of data.
3.) click "Edit Record"
4.) Observe the UI/UX changes - addition of radio fields
5.) Make a change you'd like, then navigate to bottom of change to save that change.
6.) You should be redirected back to that VA record's view, and the changes should be present.
